File: slSig/ConnectionFramework/ConnectionBase.cs
using DAL;

namespace ConnectionFramework
{
    public abstract class ConnectionBase
    {
        private System.Data.SqlClient.SqlConnection _cnn;

        public System.Data.SqlClient.SqlConnection DBConnection
        {
            get
            {
                if (_cnn == null)
                    _cnn = new System.Data.SqlClient.SqlConnection(DAL.DALBase.DBConnectionString);

                return _cnn;
            }
        }
    }
}

File: slSig/ConnectionFramework_xico/ConnectionFramework/ConnectionBase.vb
Imports System.Configuration
Imports DAL

Public MustInherit Class ConnectionBase

#Region "Atributos"

    Private _cnn As SqlClient.SqlConnection

#End Region

#Region "Propriedades"

#Region "Connections"

    Public ReadOnly Property DBConnection() As SqlClient.SqlConnection

        Get

            If IsNothing(_cnn) Then

                _cnn = New SqlClient.SqlConnection(DBConnectionString)

            End If

            Return _cnn

        End Get

    End Property

#End Region

#Region "Strings"

    Private ReadOnly Property DBConnectionString() As String

        Get

            Return DAL.ConfigConnection.GetConnection

        End Get

    End Property

#End Region

#End Region


End Class
